sql/driver/doc.go

```diff
@@ -0,0 +1,284 @@
+	// the following sql.Open call uses sc2
+	err = driver.SetSerializationConfig(sc2)
+	db4, err := sql.Open("hazelcast", "")
+
+Creating a Driver Instance Using driver.Open
+
+It is possible to create a driver instance using an existing Hazelcast client configuration using the driver.Open function.
+All client configuration items, except listeners are supported.
+
+	cfg := hazelcast.Config{}
+	cfg.Cluster.Name = "prod"
+	cfg.Serialization.SetPortableFactories(&MyPortableFactory{})
+	db := driver.Open(cfg)
+
+Executing Queries
+
+database/sql package supports two kinds of queries: The ones returning rows (select statements and a few others) and the rest (insert, update, etc.).
+The former kinds of queries are executed with QueryXXX methods and the latter ones are executed with ExecXXX methods of the sql.DB instance returned from sql.Open or driver.Open
+
+Use the question mark (?) for placeholders.
+
+Here is an Exec example:
+
+	q := `INSERT INTO person(__key, age, name) VALUES (?, ?, ?)`
+	result, err := db.Exec(q, 1001, 35, "Jane Doe")
+	// handle the error
+	cnt, err := result.AffectedRows()
+	// handle the error
+	fmt.Printf("Affected rows: %d\n", cnt)
+
+Note that LastInsertId is not supported and at the moment AffectedRows always returns 0.
+
+An example Query call:
+
+	q :=`SELECT name, age FROM person WHERE age >= ?`
+	rows, err := db.Query(q, 30)
+	// handle the error
+	defer rows.Close()
+	var name string
+	var age int
+	for rows.Next() {
+		err := rows.Scan(&name, &age)
+		// handle the error
+		fmt.Println(name, age)
+	}
+
+Context variants of Query and Exec, such as QueryContext and ExecContext are fully supported.
+Context variants are used to pass Hazelcast specific parameters.
+
+Passing Hazelcast-Specific Parameters
+
+This driver supports the following extra query parameters that Hazelcast supports:
+
+	- Cursor buffer size: Size of the server-side buffer for rows.
+	- Timeout: Maximum time a query is allowed to execute.
+
+Checkout the documentation below for details.
+
+The extra query parameters are passed in a context augmented using WithCursorBufferSize and WithQueryTimeout functions. Here is an example:
+
+	// set the cursor buffer size to 10_000
+	ctx := driver.WithCursorBufferSize(context.Background(), 10_000)
+	// set the query timeout to 2 minutes
+	ctx = driver.WithQueryTimeout(ctx, 2*time.Minute)
+	// use the parameters above with any methods that uses that context
+	rows, err := db.QueryContext(ctx, "select * from people")
+
+Creating a Mapping
+
+To connect to a data source and query it as if it is a table, a mappings should be created.
+Currently, mappings for Map, Kafka and file data sources are supported.
+
+You can read the details about mappings here: https://docs.hazelcast.com/hazelcast/latest/sql/sql-overview#mappings
+
+Supported Data Types
+
+The following data types are supported, as both builtin values and in types supporting the portable interface when inserting/updating:
```

Comment:
I think this part can be clarified. I think you mean you list builtin types and their corresponding names in portable. Actually real, decimal, date etc. are not portable names, they are hazelcast's SQL data types, aren't they?

Reply:
That's right. I'll try to clarify that.

Reply:
Updated at: https://github.com/hazelcast/hazelcast-go-client/pull/709/commits/b0798c9d652c2a57683c730b872d5d2fc19d71c0